lib/api/v1/assignment.rb

```diff
@@ -391,6 +401,19 @@ def assignment_json(assignment, user, session, opts = {})
     hash
   end
 
+  def can_show_statistics?(assignment, submission, user)
+    # To show statistics, the student's grade must be visible to the student
+    return false unless (submission.score.present? && !submission.hide_grade_from_student?)
+
+    # and the assignment must have points_possible > 0,
+    return false unless (assignment.points_possible.present? && assignment.points_possible > 0)
+
+    # Students can only see statistics when count >= 5 and not disabled by the instructor
+    # Instructor can see statistics at any time.
+    count = assignment.score_statistic&.count || 0
+    assignment.context.grants_right?(user, :read_as_admin) || (count >= 5 && !assignment.context.hide_distribution_graphs)
```

Comment:
for the invocation of this method on line 359, where we invoke this once-per-submission, it looks like these lines are going to be doing duplicated work, since it's going to be the same assignment each time this method is run.

we suggest breaking this method out into 2 separate methods: one for the submission-specific-work, and one for the assignment-specific-work. that way, the invocation on line 359 can call the submission-specific-work method once-per-submission, but only call the assignment-specific-work method one single time.